src/include/function/aggregate/min_max.h

```diff
@@ -44,41 +46,61 @@ struct MinMaxFunction {
     template<class OP>
     static inline void updatePos(uint8_t* state_, common::ValueVector* input, uint64_t multiplicity,
         uint32_t pos, storage::MemoryManager* memoryManager) {
-        updateSingleValue<OP>(reinterpret_cast<MinMaxState*>(state_), input, pos);
+        updateSingleValue<OP>(reinterpret_cast<MinMaxState*>(state_), input, pos, memoryManager);
     }
 
     template<class OP>
-    static void updateSingleValue(MinMaxState* state, common::ValueVector* input, uint32_t pos) {
+    static void updateSingleValue(MinMaxState* state, common::ValueVector* input, uint32_t pos,
+        storage::MemoryManager* memoryManager) {
         T val = input->getValue<T>(pos);
         if (state->isNull) {
-            state->val = val;
+            state->setVal(val, memoryManager);
             state->isNull = false;
         } else {
             uint8_t compare_result;
             OP::template operation<T, T>(val, state->val, compare_result);
-            state->val = compare_result ? val : state->val;
+            if (compare_result) {
+                state->setVal(val, memoryManager);
+            }
         }
     }
 
     template<class OP>
-    static void combine(uint8_t* state_, uint8_t* otherState_) {
+    static void combine(
+        uint8_t* state_, uint8_t* otherState_, storage::MemoryManager* memoryManager) {
         auto otherState = reinterpret_cast<MinMaxState*>(otherState_);
         if (otherState->isNull) {
             return;
         }
         auto state = reinterpret_cast<MinMaxState*>(state_);
         if (state->isNull) {
-            state->val = otherState->val;
+            state->setVal(otherState->val, memoryManager);
             state->isNull = false;
         } else {
             uint8_t compareResult;
             OP::template operation<T, T>(otherState->val, state->val, compareResult);
-            state->val = compareResult == 1 ? otherState->val : state->val;
+            if (compareResult) {
+                state->setVal(otherState->val, memoryManager);
+            }
         }
     }
 
     static void finalize(uint8_t* state_) {}
 };
 
+template<>
+void MinMaxFunction<common::ku_string_t>::MinMaxState::setVal(
+    common::ku_string_t& val_, storage::MemoryManager* memoryManager) {
+    if (overflowBuffer == nullptr) {
```

Comment:
Would it be better to create overflowBuffer at init state?

Reply:
Similar to collect, we don't have access to memory manager at the init state.
```
    AggregateFunction(aggr_initialize_function_t initializeFunc,
        aggr_update_all_function_t updateAllFunc, aggr_update_pos_function_t updatePosFunc,
        aggr_combine_function_t combineFunc, aggr_finalize_function_t finalizeFunc,
        common::DataType inputDataType, bool isDistinct = false)
        : initializeFunc{std::move(initializeFunc)}, updateAllFunc{std::move(updateAllFunc)},
          updatePosFunc{std::move(updatePosFunc)}, combineFunc{std::move(combineFunc)},
          finalizeFunc{std::move(finalizeFunc)}, inputDataType{std::move(inputDataType)},
          isDistinct{isDistinct} {
        initialNullAggregateState = createInitialNullAggregateState();
    }
```